### PR TITLE
bugfix: aws_ec2_placement_group spread level

### PR DESCRIPTION
**Pull request by bennylu2**
### Description
As mentioned in the [bug](https://github.com/hashicorp/terraform-provider-aws/issues/26958) that was reported, the new `spread_level` argument forced replacements. This was because the API returns `rack` as a default value when `strategy` is set to `spread`.

- Fixed up documentation
- Removed double-quotes to standardize

### Relations
Closes [#26958](https://github.com/hashicorp/terraform-provider-aws/issues/26958)

### Tests
```
make testacc TESTS='TestAccEC2PlacementGroup_spreadLevel' PKG=ec2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 20 -run='TestAccEC2PlacementGroup_spreadLevel'  -timeout 180m
=== RUN   TestAccEC2PlacementGroup_spreadLevel
=== PAUSE TestAccEC2PlacementGroup_spreadLevel
=== CONT  TestAccEC2PlacementGroup_spreadLevel
--- PASS: TestAccEC2PlacementGroup_spreadLevel (33.14s)
PASS
ok      github.com/hashicorp/terraform-provider-aws/internal/service/ec2        50.558s
```

### CODE REVIEW

File: internal/service/ec2/ec2_placement_group.go [hunk 1]
```diff
@@ -210,5 +211,11 @@ func resourcePlacementGroupCustomizeDiff(_ context.Context, diff *schema.Resourc
 		}
 	}
 
+	if diff.Id() == "" {
+		if spreadLevel, strategy := diff.Get("spread_level").(string), diff.Get("strategy").(string); spreadLevel == "" && strategy == ec2.PlacementGroupStrategySpread {
+			return diff.Clear("spread_level")
+		}
+	}
```
Comment (bennylu2) [inline internal/service/ec2/ec2_placement_group.go:218]: Clearing the diff if `spread_level` is not set but `strategy` is set to `spread`, wasn't sure if this was best way to achieve desired outcome or if there's a better way. Still getting feet wet with this

Open to suggestions, improvements, and feedback. Thanks!

**Comment by bennylu2** [comment]
Is there a way to run/limit the linters locally to only files that have had changes? Something like `pre-commit` or being able to specify a path? I feel like I may be missing something unless everyone's supposed to lint the entire codebase each time

File: internal/service/ec2/ec2_placement_group.go [hunk 2]
```diff
@@ -52,6 +52,7 @@ func ResourcePlacementGroup() *schema.Resource {
 			},
 			"spread_level": {
 				Type:         schema.TypeString,
+				Computed:     true,
```
Comment (bennylu2) [inline internal/service/ec2/ec2_placement_group.go:55]: Not sure if this is correct way to fix bug

Alternative I thought of was to set in read function but that seemed off to me as well

File: internal/service/ec2/ec2_placement_group.go (hunk 1 above)
Comment (ewbankkit) [reply internal/service/ec2/ec2_placement_group.go:218]: This code isn't necessary. Setting `spread_level` to _Computed_ is sufficient.

**Review by ewbankkit — APPROVED** [review]
LGTM :rocket:.

```console
% make testacc TESTARGS='-run=TestAccEC2PlacementGroup_' PKG=ec2 ACCTEST_PARALLELISM=2
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./internal/service/ec2/... -v -count 1 -parallel 2  -run=TestAccEC2PlacementGroup_ -timeout 180m
=== RUN   TestAccEC2PlacementGroup_basic
=== PAUSE TestAccEC2PlacementGroup_basic
=== RUN   TestAccEC2PlacementGroup_disappears
=== PAUSE TestAccEC2PlacementGroup_disappears
=== RUN   TestAccEC2PlacementGroup_tags
=== PAUSE TestAccEC2PlacementGroup_tags
=== RUN   TestAccEC2PlacementGroup_partitionCount
=== PAUSE TestAccEC2PlacementGroup_partitionCount
=== RUN   TestAccEC2PlacementGroup_defaultSpreadLevel
=== PAUSE TestAccEC2PlacementGroup_defaultSpreadLevel
=== RUN   TestAccEC2PlacementGroup_spreadLevel
=== PAUSE TestAccEC2PlacementGroup_spreadLevel
=== CONT  TestAccEC2PlacementGroup_basic
=== CONT  TestAccEC2PlacementGroup_partitionCount
--- PASS: TestAccEC2PlacementGroup_basic (24.74s)
=== CONT  TestAccEC2PlacementGroup_tags
--- PASS: TestAccEC2PlacementGroup_partitionCount (24.84s)
=== CONT  TestAccEC2PlacementGroup_disappears
--- PASS: TestAccEC2PlacementGroup_disappears (17.06s)
=== CONT  TestAccEC2PlacementGroup_spreadLevel
--- PASS: TestAccEC2PlacementGroup_spreadLevel (22.89s)
=== CONT  TestAccEC2PlacementGroup_defaultSpreadLevel
--- PASS: TestAccEC2PlacementGroup_tags (53.49s)
--- PASS: TestAccEC2PlacementGroup_defaultSpreadLevel (22.92s)
PASS
ok  	github.com/hashicorp/terraform-provider-aws/internal/service/ec2	97.809s
```

**Comment by ewbankkit** [comment]
@bennylu2 Thanks for the contribution :tada: :clap:.

**Pull request merged**